test/cppsuite/test_harness/workload_generator.h

```diff
@@ -94,17 +98,23 @@ class workload_generator : public component {
         /* Open a cursor on each collection and use the configuration to insert key/value pairs. */
         testutil_check(_config->get_int(KEY_COUNT, key_count));
         testutil_check(_config->get_int(VALUE_SIZE, value_size));
+        testutil_assert(value_size >= 0);
         for (const auto &collection_name : _collection_names) {
             /* WiredTiger lets you open a cursor on a collection using the same pointer. When a
              * session is closed, WiredTiger APIs close the cursors too. */
             testutil_check(
               session->open_cursor(session, collection_name.c_str(), NULL, NULL, &cursor));
             for (size_t j = 0; j < key_count; ++j) {
-                /* Generation of a random string value using the size defined in the test
-                 * configuration. */
-                std::string generated_value =
+                /*
+                 * Generation of a random string value using the size defined in the test
+                 * configuration.
+                 */
+                generated_value =
                   random_generator::random_generator::instance().generate_string(value_size);
-                testutil_check(insert(cursor, collection_name, j + 1, generated_value.c_str()));
+                testutil_check(session->begin_transaction(session, NULL));
+                ts = _timestamp_manager->get_next_ts();
+                testutil_check(insert(cursor, collection_name, j + 1, generated_value.c_str(), ts));
+                testutil_check(session->commit_transaction(session, NULL));
```

Comment:
We're going to need to apply the timestamp here if transactions are enabled.

Reply:
And if transactions aren't enabled don't call begin or commit.

Reply:
Add a call to _timestamp_manager->timestamps_enabled() which returns a bool, save it further up and wrap the session begin and session commit calls in the (timestamps_enabled) boolean.

When you call session->commit_transaction, if timestamps are enabled then the config is session->commit_transaction(session, "commit_timestamp=" + ts.hex()).